### PR TITLE
[19.07] travelmate: minimal change to fix cp detection

### PR DESCRIPTION
**Pull request by dibdot**
Maintainer: me / @dibdot
Compile tested: -
Run tested: -

Description:
* fix cp detection proposed by @ChristianKuehnel
* add/adapt mikrotik login script provided by @ChristianKuehnel

Signed-off-by: Dirk Brenken <dev@brenken.org>

### CODE REVIEW

**Comment by dibdot** [comment]
@ChristianKuehnel could you please test this slightly modified approach. I have no opportunity to test this due to lack of any 19.07 installation. Thanks!

**Comment by ChristianKuehnel** [comment]
This is related to #12867

File: net/travelmate/files/travelmate.sh [hunk 1]
```diff
@@ -228,7 +228,13 @@ f_net()
...
+	raw="$(${trm_fetch} --timeout=$((trm_maxwait/6)) "${trm_captiveurl}" -O /dev/null 2>&1)"
+	if [ "$(printf "%s" "${raw}" | grep -c "^Redirected")" = "1" ] 
+	then
+		raw="$(printf "%s" "${raw}" | grep "^Redirected")"
+	else
+		raw="$(printf "%s" "${raw}" | tail -n 1)"
+	fi
 	raw="$(printf "%s" "${raw//[\?\$\%\&\+\|\'\"\:\*\=\/]/ }")"
 	result="$(printf "%s" "${raw}" | awk '/^Failed to redirect|^Redirected/{printf "%s","net cp";exit}/^Download completed/{printf "%s","net ok";exit}/^Failed|Connection error/{printf "%s","net nok";exit}')"
 	if [ "${result}" = "net cp" ]
```
Comment (ChristianKuehnel) [inline net/travelmate/files/travelmate.sh:240]: Nope, does not work. Not sure what line 242 is supposed to do, but it does not work with 
```bash
raw="Redirected to /login?dst=http%3A%2F%2Fcaptive.apple.com%2F on 172.16.0.1"
```

This is why I replaced that line in my proposal with a simpler version that just add the domain name after the `on`.
```
result="$(printf "%s" "net cp : ${raw//*on /}")"
```

**Comment by dibdot** [comment]
Should work now - please test again. Your proposed solution does not separate the cp domain properly if the "on" is followed not only by the domain/IP, e.g. "hotspot.vodafone.de/index.html"

**Review by ChristianKuehnel — APPROVED** [review]
Yes, this works. 👍 Thank you!

**Comment by dibdot** [comment]
Many thanks for testing! I'll keep this PR open until your login script has been finished ... it's also doable that I add your login script to this PR (just in case).

Thanks again fro your work!

**Pull request merged**